core/consensus/src/synchronization.rs

```diff
@@ -82,9 +97,25 @@ where
             self.check_proof(id, proof.clone())?;
 
             if id != 1 && current_hash != epoch.header.pre_hash {
+                trace::error(
+                    "prev_hash_diff".to_string(),
+                    Some(json!({
+                        "epoch_id": id,
+                        "current_prev_hash": epoch.header.pre_hash.as_hex(),
```

Comment:
If this should be `current_hash.as_hex()`

Reply:
Why? This means the `prev_hash` are different.